source/hardware/controllers/pioneer_ddj_sb3.rst

```diff
@@ -0,0 +1,381 @@
+10   :hwlabel:`1/2X`                                    Halve loop length
+11   Pad 1 - 8                                          Perform different functions depending on current mode
+12   :hwlabel:`PLAY/PAUSE` button                       Play / pause
+12   :hwlabel:`SHIFT` + :hwlabel:`PLAY/PAUSE` button    Return to temporary cue point
+13   :hwlabel:`CUE` button                              Set, play and call out cue points
+13   :hwlabel:`SHIFT` + :hwlabel:`CUE` button           Return to cue point and stop or load previous track
+14   :hwlabel:`SYNC` button                             Match tempo and phase of adjacent deck
+14   :hwlabel:`SHIFT` + :hwlabel:`SYNC` button          Turn off sync mode
+15   :hwlabel:`SHIFT` + button                          Switch function of controls
+16   :hwlabel:`DECK X`                                  Switch deck
+17   :hwlabel:`KEY LOCK`                                Toggle key lock
+17   :hwlabel:`SHIFT` + :hwlabel:`KEY LOCK`             Toggle tempo slider range
+===  =================================================  ============================================================================================
+
+
+.. _pioneer-ddj-sb3-mixer:
+
+Mixer section (p. 8)
+~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../../_static/controllers/pioneer_ddj_sb3_mixer.svg
+   :align: center
+   :width: 50%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-SB3 (mixer section)
+   :figclass: pretty-figures
+
+   Mixer section (p. 8)
+
+===  =====================================================  ============================================================================================
+No.  Control                                                Function
+===  =====================================================  ============================================================================================
+1    :hwlabel:`TRIM` knob                                   Adjust individual channel output gain
+2    :hwlabel:`HI`/:hwlabel:`MID`/:hwlabel:`LOW` knobs      Adjust high/mid/low-frequencies
+3    :hwlabel:`FILTER` knob                                 Low pass / high pass filter for each channel
+4    :hwlabel:`HEADPHONE CUE 1` button                      Toggle headphone pre-fader listening of left deck.
+4    :hwlabel:`HEADPHONE CUE 2` button                      Toggle headphone pre-fader listening of right deck.
+5    Channel faders                                         Adjust the output level for each channel
+5    :hwlabel:`SHIFT` Channel faders                        Fader start and adjust the output level for each channel
+6    Cross fader                                            Fade between left and right deck
+7    :hwlabel:`MASTER` knob                                 Adjust master output level
+8    :hwlabel:`HEADPHONES` knob                             Adjust headphones output level
+9    :hwlabel:`MASTER` button                               Output master to headphones
```

Comment:
Do these affect the hardware's audio interface or Mixxx's software knobs? Please document that. https://github.com/mixxxdj/mixxx/wiki/Contributing-Mappings#main--headphone-gain-knobs

Reply:
I'm not sure how to answer this. The controller does send MIDI messages when the knobs are moved. The volume changes regardless of whether you move the knob or the software button. I think the answer is both but you'll have to let me know what needs documenting. Is there an example?

Reply:
If they change the volume of the audio interface they should not be mapped as described on the wiki.

Reply:
So if I turn the knob on the controller, it's ok if the knob doesn't move on the screen? Just want to confirm I understand.

Reply:
Yes, otherwise you would change two gain stages with a single knob (the software gain in mixxx and the hardware gain on the controller) with a single knob. That usually makes it very hard to actually control the gain because the double scaling makes it very sensitive.